### PR TITLE
[xdoctest][task 335] reformat example code with google style in python/paddle/base/dygraph/tensor_patch_methods.py

### PR DESCRIPTION
**Pull request by yoyoIcy**
### PR types

Others

### PR changes

Others

### Description

修改如下文件的示例代码，使其通过 `xdoctest` 检查：

- `python/paddle/base/dygraph/tensor_patch_methods.py`

预览：

- en: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/en/api/paddle/jit/XXX_en.html
- zh: http://preview-paddle-pr-XXX.paddle-docs-preview.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/jit/XXX_cn.html

### Related links
 
- #55629
- #55295

@sunzhongkai588 @SigureMo @megemini

### CODE REVIEW

File: python/paddle/base/dygraph/tensor_patch_methods.py [hunk 1]
```diff
@@ -252,32 +250,32 @@ def backward(self, grad_tensor=None, retain_graph=False):
...
+                >>> grad_tensor=paddle.to_tensor(2.)
+                >>> for i in range(5):
+                ...     y = paddle.pow(x, 4.0)
+                ...     y.backward(grad_tensor)
+                ...     print("{}: {}".format(i, x.grad))
+                0: [1000.]
+                1: [2000.]
+                2: [3000.]
+                3: [4000.]
+                4: [5000.]
```
Comment (megemini) [inline python/paddle/base/dygraph/tensor_patch_methods.py]: 输出是 Tensor

File: python/paddle/base/dygraph/tensor_patch_methods.py [hunk 2]
```diff
@@ -334,13 +332,13 @@ def gradient(self):
...
-                x = paddle.to_tensor(5., stop_gradient=False)
-                y = paddle.pow(x, 4.0)
-                y.backward()
-                print("grad of x: {}".format(x.gradient()))
-                # [500.]
+                >>> x = paddle.to_tensor(5., stop_gradient=False)
+                >>> y = paddle.pow(x, 4.0)
+                >>> y.backward()
+                >>> print("grad of x: {}".format(x.gradient()))
+                [500.]
```
Comment (megemini) [inline python/paddle/base/dygraph/tensor_patch_methods.py]: 输出不对～

File: python/paddle/base/dygraph/tensor_patch_methods.py [hunk 3]
```diff
@@ -372,41 +370,44 @@ def register_hook(self, hook):
...
+                >>> # print_hook_fn print content in backward
+                Tensor(shape=[4], dtype=float32, place=CUDAPlace(0), stop_gradient=False,
+                [2., 4., 6., 8.])
+
+                >>> print("w.grad:", w.grad)
+                w.grad: [1. 2. 3. 4.]
+                >>> print("x.grad:", x.grad)
+                x.grad: [ 4.  8. 12. 16.]
+                >>> print("y.grad:", y.grad)
+                y.grad: [2. 4. 6. 8.]
```
Comment (megemini) [inline python/paddle/base/dygraph/tensor_patch_methods.py]: 检查一下输出

File: python/paddle/base/dygraph/tensor_patch_methods.py [hunk 4]
```diff
@@ -622,13 +635,13 @@ def __str__(self):
...
+                >>> import paddle
+                >>> x = paddle.rand([2, 5])
+                >>> print(x)
 
-                # Tensor(shape=[2, 5], dtype=float32, place=CPUPlace,
-                #        [[0.30574632, 0.55739117, 0.30902600, 0.39413780, 0.44830436],
-                #         [0.79010487, 0.53972793, 0.09495186, 0.44267157, 0.72112119]])
+                Tensor(shape=[2, 5], dtype=float32, place=CPUPlace,
+                [[0.30574632, 0.55739117, 0.30902600, 0.39413780, 0.44830436],
+                 [0.79010487, 0.53972793, 0.09495186, 0.44267157, 0.72112119]])
```
Comment (megemini) [inline python/paddle/base/dygraph/tensor_patch_methods.py]: 加 seed ～ 用 cpu place ～

File: python/paddle/base/dygraph/tensor_patch_methods.py [hunk 5]
```diff
@@ -641,19 +654,18 @@ def __deepcopy__(self, memo):
...
-                print(y)
-                # Tensor(shape=[], dtype=float32, place=CPUPlace, stop_gradient=True,
-                #        2.)
+                >>> print(x)
+                Tensor(shape=[], dtype=float32, place=CPUPlace, stop_gradient=True,
+                2.)
 
+                >>> print(y)
+                Tensor(shape=[], dtype=float32, place=CPUPlace, stop_gradient=True,
+                2.)
```
Comment (megemini) [inline python/paddle/base/dygraph/tensor_patch_methods.py]: place=Place(cpu)

**Review by SigureMo — APPROVED** [review]
LGTMeow 🐾

**Pull request merged**